client/js/table-view.js

```diff
@@ -1,103 +1,120 @@
 	}
+
 	init(){
-		this.initDomeReferences();
+		this.initDomReferences();
 		this.initCurrentCell();
 		this.renderTable();
 		this.attachEventHandlers();
 	}
+
 	normalizeValueForRendering(value){
-		return value || "";
+		return value || '';
 	}
 
 	renderFormulaBar(){
 		const currenCellValue = this.model.getValue(this.currentCellLocation);
 		this.formulaBarEl.value = this.normalizeValueForRendering(currenCellValue); 
 		this.formulaBarEl.focus();
-
 	}
+
 	initCurrentCell(){
 		this.currentCellLocation = {col: 0, row: 0};
 		this.renderFormulaBar();
 	}
 
 	attachEventHandlers(){
-		this.sheetBodyEl.addEventListener("click", this.handleSheetClick.bind(this));
-		this.formulaBarEl.addEventListener("keyup", this.handleFormulaBarChange.bind(this));
+		this.sheetBodyEl.addEventListener('click', this.handleSheetClick.bind(this));
+		this.formulaBarEl.addEventListener('keyup', this.handleFormulaBarChange.bind(this));
+		this.addRow.addEventListener('click', this.addRowButton.bind(this));
+		this.addCol.addEventListener('click', this.addColButton.bind(this));
+	}
+
+	addColButton(){
+		this.model.addColumn();
+		this.renderTable();
+	}
+
+	addRowButton(){
+		this.model.addRow();
+		this.renderTable();
 	}
 	
 	handleFormulaBarChange(evt){
 		const value = this.formulaBarEl.value;
 		this.model.setValue(this.currentCellLocation, value);
 		this.renderTableBody();
-		this.getSum(value);
-		
+		this.renderTableFooter();
 	}
 
-	getSum(value){
-		console.log(value);
-		console.log("col ->"+this.currentCellLocation.col);
-		console.log("row ->"+this.currentCellLocation.row);
-		const sumLocation = {col: this.currentCellLocation.col, row: 19};
-		let arr = Array(9).fill(0);
-		let arrIndex = sumLocation.col
-		arr[arrIndex] = parseInt(arr[arrIndex])+parseInt(value);
-		console.log(arr[arrIndex]);
-		this.model.setValue(sumLocation, arr[sumLocation.col]);
-
-	}
-
-
 	handleSheetClick(evt){
 		const col = evt.target.cellIndex;
 		const row = evt.target.parentElement.rowIndex -1;
-
 		this.currentCellLocation = {col: col, row: row};
 		this.renderTableBody();
 		this.renderFormulaBar();
 	}
 
-
-	initDomeReferences(){
-		this.headerRowEl = document.querySelector("THEAD TR");
-		this.sheetBodyEl = document.querySelector("TBODY");
-		this.formulaBarEl = document.querySelector("#formula-bar");
+	initDomReferences(){
+		this.headerRowEl  = document.querySelector('THEAD TR');
+		this.sheetBodyEl  = document.querySelector('TBODY');
+		this.formulaBarEl = document.querySelector('#formula-bar');
+		this.footerRowEl  = document.querySelector('TFOOT TR');
+		this.addRow       = document.querySelector('#addRow');
+		this.addCol       = document.querySelector('#addCol');
 	}
 
 	renderTable(){
 		this.renderTableHeader();
 		this.renderTableBody();
+		this.renderTableFooter();
 	}
 
 	renderTableHeader(){
 		removeChildren(this.headerRowEl); 
-		getLetterRange("A", this.model.numCols)
+		getLetterRange('A', this.model.numCols)
 		.map(colLebel => createTH(colLebel))
```

Comment:
Please indent the .map() and .forEach() methods one additional tab (2 spaces). This is the convention for chained methods.